field/bitmap.go

```diff
@@ -34,82 +38,74 @@ func (f *Bitmap) SetSpec(spec *Spec) {
 }
 
 func (f *Bitmap) SetBytes(b []byte) error {
-	f.bitmap = utils.NewBitmapFromData(b)
-	if f.data != nil {
-		*(f.data) = *f
-	}
+	f.data = b
 	return nil
 }
 
 func (f *Bitmap) Bytes() ([]byte, error) {
 	if f == nil {
 		return nil, nil
 	}
-	return f.bitmap.Bytes(), nil
+	return f.data, nil
 }
 
 func (f *Bitmap) String() (string, error) {
 	if f == nil {
 		return "", nil
 	}
-	return f.bitmap.String(), nil
-}
-
-func (f *Bitmap) Pack() ([]byte, error) {
-	f.setBitmapFields()
 
-	count := f.bitmapsCount()
+	var bits []string
 
-	// here we have max possible bytes for the bitmap 8*maxBitmaps
-	data, err := f.Bytes()
-	if err != nil {
-		return nil, fmt.Errorf("failed to retrieve bytes: %w", err)
+	for _, byte_ := range f.data {
+		bits = append(bits, fmt.Sprintf("%08b", byte_))
 	}
 
-	data = data[0 : 8*count]
+	return strings.Join(bits, " "), nil
+}
 
-	packed, err := f.spec.Enc.Encode(data)
+func (f *Bitmap) Pack() ([]byte, error) {
+	packed, err := f.spec.Enc.Encode(f.data)
 	if err != nil {
 		return nil, fmt.Errorf("failed to encode content: %w", err)
 	}
 
 	return packed, nil
 }
 
-// Unpack of the Bitmap field returns data of varied length
-// if there is only primary bitmap (bit 1 is not set) we return only 8 bytes (or 16 for hex encoding)
-// if secondary bitmap presents (bit 1 is set) we return 16 bytes (or 32 for hex encoding)
-// and so on for maxBitmaps
+// Unpack sets the bitmap data. It returns the number of bytes read from the
+// data. Usually it's 8 for binary, 16 for hex - for a single bitmap.
+// If DisableAutoExpand is not set (default), it will read all bitmaps until
+// the first bit of the read bitmap is not set.
+// If DisableAutoExpand is set, it will only read the first bitmap regardless
+// of the first bit being set.
 func (f *Bitmap) Unpack(data []byte) (int, error) {
-	minLen, _, err := f.spec.Pref.DecodeLength(minBitmapLength, data)
+	minLen, _, err := f.spec.Pref.DecodeLength(f.bitmapLenght, data)
 	if err != nil {
 		return 0, fmt.Errorf("failed to decode length: %w", err)
 	}
 
-	rawBitmap := make([]byte, 0)
+	f.data = make([]byte, 0)
 	read := 0
 
-	// read max
-	for i := 0; i < maxBitmaps; i++ {
+	var i int
+
+	// read until we have no more bitmaps
+	// or only read one bitmap if DisableAutoExpand is set
+	for {
+		i++
 		decoded, readDecoded, err := f.spec.Enc.Decode(data[read:], minLen)
 		if err != nil {
-			return 0, fmt.Errorf("failed to decode content for %d bitmap: %w", i+1, err)
+			return 0, fmt.Errorf("failed to decode content for %d bitmap: %w", i, err)
 		}
 		read += readDecoded
+		f.data = append(f.data, decoded...)
 
-		rawBitmap = append(rawBitmap, decoded...)
-		bitmap := utils.NewBitmapFromData(decoded)
-
-		// if no more bitmaps, exit loop
-		if !bitmap.IsSet(1) {
+		// if it's a fixed bitmap or first bit of the decoded bitmap is not set, exit loop
+		if f.spec.DisableAutoExpand || decoded[0]&(1<<7) == 0 {
```

Comment:
should I use `128` instead of `1<<7` as I mix these both approaches in the code? Want to be more consistent.